add ResetCheckpoints function

frrad opened the pull request:
Adds a `ResetCheckpoints` function which sets all existing dynamo checkpoints to `LATEST`. 

This is useful to call as part of a deploy if you want to make sure that all new readers start at the most recent record instead of picking up where the the old readers left off.

Commit:
add ResetCheckpoints function

garethlewin commented on kinsumer.go:
```diff
@@ -173,9 +174,9 @@ func (k *Kinsumer) refreshShards() (bool, error) {
 		return false, err
 	}
 
-	changed := (totalClients != k.totalClients) ||
```
Not sure these changes are warranted, I added the brackets for readability.

garethlewin commented on kinsumer.go:
```diff
@@ -483,6 +484,51 @@ func (k *Kinsumer) CreateRequiredTables() error {
 }
 
+func (k *Kinsumer) ResetCheckpoints(sess *session.Session) error {
+	entries, err := k.dynamodb.Scan(&dynamodb.ScanInput{TableName: aws.String(k.checkpointTableName)})
```
Probably better to use `ScanPages`, some kinesis streams have many many shards.

Commit:
revert `gofmt -s` changes

garethlewin commented on kinsumer.go:
```diff
@@ -483,6 +484,51 @@ func (k *Kinsumer) CreateRequiredTables() error {
+		shard := entry["Shard"].S
+
+		if shard == nil {
+			return fmt.Errorf("found %v for Shard entry", entry["Shard"].S)
```
Probably fine to use "%s" here and then let the AttributeValue make it pretty https://github.com/aws/aws-sdk-go/blob/master/service/dynamodb/api.go#L6125-L6127

garethlewin commented on kinsumer.go:
```diff
@@ -483,6 +484,51 @@ func (k *Kinsumer) CreateRequiredTables() error {
+	if err != nil {
+		return err
+	}
+	_, err = k.dynamodb.UpdateItem(&dynamodb.UpdateItemInput{
```
For a large table this could get slow. Especially since you are reusing the same expression (just difference condition) for each update. Also there is no `batchUpdateItem` so maybe we should consider going wide. Probably fine though.

Commit:
switch to using ScanPages

Commit:
use %s in Errorf

frrad commented:
Addressed your comments. Please take another look.

For my use case I am not slowed down too much by doing `updateItem` in just one goroutine, but happy to add more if you want. Let me know.

Commit:
remove superfluous variable
